src/librustc_driver/driver.rs

```diff
@@ -233,24 +242,14 @@ pub fn compile_input(sess: &Session,
                 }
             }
 
-            #[cfg(not(feature="llvm"))]
-            {
-                let _ = incremental_hashes_map;
-                sess.err(&format!("LLVM is not supported by this rustc"));
-                sess.abort_if_errors();
-                unreachable!();
-            }
-
-            #[cfg(feature="llvm")]
             Ok((outputs, trans))
         })??
     };
 
     #[cfg(not(feature="llvm"))]
```

Comment:
Is this `#[cfg]` still necessary with the lint allow directive above? Maybe it could just be a `if cfg!`?